tests/dir_helpers.py

```diff
@@ -200,6 +200,20 @@ def branch(self, name, new=False):
         finally:
             self.scm.checkout(old)
 
+    @contextmanager
+    def local_remote(self):
```

Comment:
@pared, this as contextmanager doesnot really make sense. It'd be better to have a explicit `push` on tests as well.

Also, we have a `local_remote` fixture already below.

Reply:
fixing it RN

Reply:
@skshetry, a new iteration of this issue has been pushed, I was able to remove most of the remote setups, though I am not sure whether this context manager should exist. What do you think?

Reply:
@pared, I'd love to have explicit push.
What about something like this:
```python
def test(tmp_dir, dvc, setup_remote):
# setup_remote fixture provides a function with which we can setup remote.
setup_remote(dvc, url='', name='', default=True)
# if url is not provided, create a tmp_dir and set that as a local remote
```

Reply:
It feels like the commands in `dvc remote` should really be part of `Repo()` api.

Reply:
@skshetry I introduced `setup_remote`